### ISSUE TITLE
bus_type = 'virtio' doesn't create a controller

### ISSUE THREAD

**Issue by ladar** [vagrant-libvirt/vagrant-libvirt#1054]
### Steps to reproduce

For some reason setting `bus_type = "virtio"` works properly on some VMs, like `generic/debian10` but not with others, like `generic/debian9`.  Note that for some reason `generic/debian8` works with `virtio`. 

Setting `bus_type = "scsi"` works with those boxes, like `generic/debian9` but appears to cause problems with others, like `generic/centos8`.   

From what I could tell by looking at the raw XML, when the bus type is `scsi`, it forces the creation of a `virtio-scsi` controller. While a bus type of `virtio` does not create a disk controller of any kind.  

### Expected behaviour

From what I could gather by looking at the libvirt documentation, a bus type of `virtio` should create a controller with a type `virtio` and a model of `auto` (or perhaps `virtio-scsi`.

### Actual behaviour

The box fails to boot properly without a disk controller. 

It may be possible to mitigate this issue by modifying the kernel image inside the box file. If anyone  knows what's required, please let me know.

### System configuration

Tested using CentOS 7.

**Libvirt version:**

libvirtd (libvirt) 4.5.0

**Output of `vagrant version; vagrant plugin list`:**

Installed Version: 2.2.6
Latest Version: 2.2.6
 
You're running an up-to-date version of Vagrant!
vagrant-libvirt (0.0.45, global)

**Are you using upstream vagrant package or your distros package?**
Upstream

Attached are the different XML files for comparison. 

[debian9.scsi.xml.txt](https://github.com/vagrant-libvirt/vagrant-libvirt/files/3737540/debian9.scsi.xml.txt)
[debian9.virtio.xml.txt](https://github.com/vagrant-libvirt/vagrant-libvirt/files/3737541/debian9.virtio.xml.txt)
[debian10.xml.txt](https://github.com/vagrant-libvirt/vagrant-libvirt/files/3737542/debian10.xml.txt)

And sample a `Vagrantfile` for each scenario.

[Vagrantfile.debian10.virtio.txt](https://github.com/vagrant-libvirt/vagrant-libvirt/files/3737547/Vagrantfile.debian10.virtio.txt)
[Vagrantfle.debian9.scsi.txt](https://github.com/vagrant-libvirt/vagrant-libvirt/files/3737548/Vagrantfle.debian9.scsi.txt)
[Vagrantfle.debian9.virtio.txt](https://github.com/vagrant-libvirt/vagrant-libvirt/files/3737549/Vagrantfle.debian9.virtio.txt)

**Comment by ladar** [comment]
I ran test, and changed the disk bus to virtio, and about half the boxes didn't boot, so keeping the change limited to RHEL 8 and its relatives. I didn't check the console to find out why they didn't boot... and the test was run on a CentOS 7 host with libvirt 4.5.0, and qemu-kvm 2.12.0 installed. I suspect the results will vary based on host/iibvirt/qemu version.


```

The generic alpine35 libvirt test run passed!
The generic alpine36 libvirt test run passed!
The generic alpine37 libvirt test run passed!
The generic alpine38 libvirt test run passed!
The generic alpine39 libvirt test run passed!
The generic alpine310 libvirt test run passed!
The generic arch libvirt test run passed!
The generic centos6 libvirt test run passed!
The generic centos7 libvirt test run passed!
The generic centos8 libvirt test run passed!
The generic debian8 libvirt test run passed!
The generic debian9 libvirt test run failed! Exiting. { !! }
The generic debian10 libvirt test run passed!
The generic dragonflybsd5 libvirt test run failed! Exiting. { !! }
The generic fedora25 libvirt test run passed!
The generic fedora26 libvirt test run passed!
The generic fedora27 libvirt test run passed!
The generic fedora28 libvirt test run passed!
The generic fedora29 libvirt test run passed!
The generic fedora30 libvirt test run passed!
The generic freebsd11 libvirt test run failed! Exiting. { !! }
The generic freebsd12 libvirt test run failed! Exiting. { !! }
The generic gentoo libvirt test run failed! Exiting. { !! }
The generic hardenedbsd11 libvirt test run failed! Exiting. { !! }
The generic hardenedbsd12 libvirt test run failed! Exiting. { !! }
The generic netbsd8 libvirt test run failed! Exiting. { !! }
The generic openbsd6 libvirt test run passed!
The generic opensuse15 libvirt test run failed! Exiting. { !! }
The generic opensuse42 libvirt test run failed! Exiting. { !! }
The generic oracle7 libvirt test run passed!
The generic oracle8 libvirt test run passed!
The generic rhel6 libvirt test run passed!
The generic rhel7 libvirt test run passed!
The generic rhel8 libvirt test run passed!
The generic ubuntu1604 libvirt test run passed!
```

**Comment by electrofelix** [comment]
This can often depend on the kernel modules installed in the client image as well as the Vagrantfile that is packaged with the box and how the default disk was originally created. But if there is something missing that by adding it in automatically results in more VMs booting consistently that would be good

**Comment by ladar** [comment]
All of the boxes are built with the `packer`, using `"disk_interface": "virtio-scsi"` and `"net_device": "virtio-net"`.

I don't know if adding the controller explicitly will fix the problem. But I believe it was the solution when we discovered some platforms were assuming a using an emulated LSI controller, while others were using `virtio-scsi`.

What I'm more interested in is trying to make the box images work, by default, on as many platforms as possible. And that means bundling a `Vagrantfile` with the default options that work. I just don't know what that is.

Like I said above, the `scsi` option seemed to work best on the platforms I tested with ... until RHEL 8 came along and wouldn't boot using SCSI. Switching the Vagrantfile to `virtio` seems to work on my computer, but I'm worried it doesn't work properly everywhere. 

Either way, if anybody knows what kernel modules are needed, or what the `Vagrantfile` defaults should, let me know, so I can update the boxes accordingly.

I don't know if switching the disk bus to `virtio` changes the UUID, or disk device path, or it's a kernel module issue. But if it's the latter, then it's probably one of these modules:

```
CONFIG_BLK_MQ_VIRTIO=y
CONFIG_VIRTIO_VSOCKETS=m
CONFIG_VIRTIO_VSOCKETS_COMMON=m
CONFIG_VIRTIO_BLK=m
# CONFIG_VIRTIO_BLK_SCSI is not set
CONFIG_SCSI_VIRTIO=m
CONFIG_VIRTIO_NET=m
CONFIG_VIRTIO_CONSOLE=m
CONFIG_HW_RANDOM_VIRTIO=y
CONFIG_DRM_VIRTIO_GPU=m
CONFIG_VIRTIO=y
CONFIG_VIRTIO_MENU=y
CONFIG_VIRTIO_PCI=y
CONFIG_VIRTIO_PCI_LEGACY=y
CONFIG_VIRTIO_BALLOON=m
CONFIG_VIRTIO_INPUT=m
# CONFIG_VIRTIO_MMIO is not set
# CONFIG_RPMSG_VIRTIO is not set
CONFIG_CRYPTO_DEV_VIRTIO=m
```

This list was taken from a CentOS 8 machine. And if someone knows what to add, perhaps I can add to the various boxes that didn't work.

**Comment by electrofelix** [comment]
With #692 merged and now #1520, any box built by packer using `virtio-scsi` should just set the disk_bus to `scsi` in the box Vagrantfile and the next release should do the correct thing.

Going to mark this as fixed with #1520 once it lands.
